### PR TITLE
Fix process termination and add additional mknod capability

### LINKED ISSUE #3671: New upcoming Gitpod cannot run 3 docker pull during prebuild
### Update March 31: 
Previously (last night), running 2 docker pull commands during prebuild would reproduce the bug.
But this morning, running 3 docker pull commands is required to reproduce the bug.

### Bug description

In the new (upcoming) Gitpod version, running **3** `docker pull` commands during prebuild, would result in loss of docker images, and any files created during prebuild.

(Current Gitpod version doesn't have this bug, and work as expected)

### Steps to reproduce

In the new (upcoming) Gitpod version:
1. Run more than **3** `docker pull` commands during prebuild.
1. Generate a file during prebuild.
1. After prebuild, when workspace starts - no docker images and no files generated during prebuild exist.

### Expected behavior

Similar to current Gitpod version, that all docker images that were downloaded during prebuild, and all the files generated during prebuild, would exist when starting a new workspace.

### Example repository

https://github.com/shaal/gitpod-bug-repro/tree/bug/docker-pull

### PR DESCRIPTION
**Pull request by aledbf**
*How to test:*
- Open https://aledbf-killdocker.staging.gitpod-dev.com/
- Start a prebuild https://aledbf-killdocker.staging.gitpod-dev.com/#prebuild/https://github.com/shaal/gitpod-bug-repro/tree/bug/docker-pull
- ws-daemon should not contain errors restauring snapshot
- gitpod UI should look like:

Screenshot from 2021-04-01 13-20-48
Screenshot from 2021-04-01 13-21-47

fixes #3671

### CODE REVIEW

File: components/ws-daemon/pkg/iws/livebackup.go [hunk 1]
```diff
@@ -104,6 +104,7 @@ func (l *LiveWorkspaceBackup) Backup() (dest string, err error) {
 				continue
 			}
 
+			log.WithField("dir", filepath.Join(gpdir, fn)).Info("Removing directory")
```
Comment (csweichel) [inline components/ws-daemon/pkg/iws/livebackup.go:107]: leftover?

File: components/content-service/pkg/initializer/initializer.go [hunk 2]
```diff
@@ -315,7 +315,9 @@ func InitializeWorkspace(ctx context.Context, location string, remoteStorage sto
 			return src, xerrors.Errorf("cannot clean workspace folder: %w", err)
 		}
 		for _, f := range fs {
-			err := os.RemoveAll(filepath.Join(location, f.Name()))
+			path := filepath.Join(location, f.Name())
+			log.WithField("dir", path).Info("Removing directory")
```
Comment (csweichel) [inline components/content-service/pkg/initializer/initializer.go:319]: leftover?

File: components/content-service/pkg/initializer/prebuild.go [hunk 3]
```diff
@@ -126,6 +126,7 @@ func clearWorkspace(location string) error {
 		return err
 	}
 	for _, file := range files {
+		log.WithField("dir", file).Info("Removing directory")
```
Comment (csweichel) [inline components/content-service/pkg/initializer/prebuild.go:129]: leftover?

Comment (aledbf) [reply components/content-service/pkg/initializer/prebuild.go]: fixed

File: components/content-service/pkg/initializer/initializer.go (hunk 2 above)
Comment (aledbf) [reply components/content-service/pkg/initializer/initializer.go]: fixed

File: components/ws-daemon/pkg/iws/livebackup.go (hunk 1 above)
Comment (aledbf) [reply components/ws-daemon/pkg/iws/livebackup.go]: fixed

**Review by csweichel — APPROVED** [review]
works like a charm. Thank for fixing this.

**Pull request merged**